### DISCUSSION TITLE
How to detect double clicks?

### DISCUSSION THREAD

**Discussion by veniamin-ilmer** [hecrj/iced#2015] [Q&A] [answered]
I see there is a double click enum - https://docs.iced.rs/iced_core/mouse/click/enum.Kind.html#variant.Double

Yet, the mouse events seem quite limited, only detecting press and release - https://docs.iced.rs/iced_core/mouse/enum.Event.html

**Comment by AustinMReppert** [comment]
I have been messing with the mouse area widget on a fork, it might help you understand.
* It doesn't really work on touch input though.

https://github.com/AustinMReppert/iced/blob/bde02dacf8bd89978161ecba5c9e57816b295611/widget/src/mouse_area.rs#L328

**Reply by veniamin-ilmer** [reply]
This code doesn't seem to detect double clicks. I have my own custom widget which can detect clicks by observing mouse press then mouse release. The problem with a double click is it requires timing how long it look between each click. I would like to avoid writing that out if possible.

If you look at https://docs.iced.rs/src/iced_core/mouse/click.rs.html#64 it seems like there is already some functionality which attempts to time the duration between clicks to detect if it was a double click or not. But although this is part of the core code of iced, it seems like it is not tied to the mouse event for some reason.

**Reply by AustinMReppert** [reply, marked as answer]
![image](https://github.com/iced-rs/iced/assets/29180043/655d407c-8836-43fa-aa34-c832cf95c294)
If you run the tour example with debug you should be able to see the events.
